### PR TITLE
Update `bridgeTxHistory` method in TBTC wrapper

### PR DESCRIPTION
**Pull request by r-czajkowski**
Closes: #388

Take into account fees while calculating an amount. If it's a revealed deposit we need to get `treasuryFee` param from onchain `DepositRequest` struct because the `treausuryFee` depends on the deposited amount and `depositTreasuryFeeDivisor` param which is governable so it means it may change over time.

If it's an already minted deposit the only way to get the exact amount of minted `tBTC` is:
1. Get all logs from transaction in which the optimistic minting was completed.
2. Find `Transfer` events from tBTC token contract based on logs.
3. Find `Transfer` event where `to` param is equal to the depositor address. There is only one transfer to the depositor account.

### CODE REVIEW

File: src/threshold-ts/tbtc/index.ts [hunk 1]
```diff
@@ -384,33 +420,55 @@ export class TBTC implements ITBTC {
...
+      depositor,
+      depositKeys
+    )
+
+    const estimatedAmountToMintByDepositKey =
+      await this._calculateEstimatedAmountToMintForRevealedDeposits(
+        mintedDepositEvents.map((event) =>
+          (event.args?.depositKey as BigNumber).toHexString()
+        )
+      )
```
Comment (michalsmiarowski) [inline src/threshold-ts/tbtc/index.ts:433]: We should pass `depositKeys` array that we got from `revealedDeposits` not `mintedDepositEvents. We need `estimatedAmountToMintByDepositKey` to calculate the amount for all revealed deposits (that can be also in pending state).

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some comments to look at before the merge:

File: src/hooks/tbtc/useSubsribeToDepositRevealedEvent.ts [hunk 2]
```diff
@@ -31,11 +31,15 @@ export const useSubscribeToDepositRevealedEvent = () => {
...
+      const { amountToMint } = await threshold.tbtc.getEstimatedFees(
+        amount.toString()
+      )
+
       dispatch(
         tbtcSlice.actions.depositRevealed({
           fundingOutputIndex,
           fundingTxHash,
-          amount: amount.toString(),
+          amount: amountToMint,
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/hooks/tbtc/useSubsribeToDepositRevealedEvent.ts:42]: What about `amountWithoutFees`? 😃 

If not then maybe we could at least add a coment about it in the `BridgeTxHistory` interface?

File: src/threshold-ts/tbtc/index.ts (hunk 1 above)
Comment (r-czajkowski) [reply src/threshold-ts/tbtc/index.ts]: 50a511f

**Review by michalsmiarowski — APPROVED** [review]
LGTM

**Pull request merged**